clients/unshaded/src/main/java/tachyon/client/WriteType.java

```diff
@@ -15,33 +15,43 @@
 
 package tachyon.client;
 
+import tachyon.annotation.PublicApi;
+
 /**
- * Different write types for a TachyonFile.
+ * Convenience modes for commonly used write types for a TachyonFile.
  *
- * As of 0.8, replaced by {@link TachyonStorageType} and {@link UnderStorageType}
+ * For finer grained control over data storage, advanced users may specify
+ * {@link tachyon.client.TachyonStorageType} and {@link tachyon.client.UnderStorageType}.
  */
-@Deprecated
+@PublicApi
 public enum WriteType {
   /**
-   * Write the file and must cache it.
+   * Write the file and must cache it. The data will be written to the highest tier in a worker's
```

Comment:
define 'must cache', e.g. 'Write the file and cache it. If caching fails, throw an exception'